src/RuleMatcher.h

```diff
@@ -168,6 +169,35 @@ class RuleEndpointState {
     RuleEndpointState(analyzer::Analyzer* arg_analyzer, bool arg_is_orig, RuleEndpointState* arg_opposite,
                       analyzer::pia::PIA* arg_PIA);
 
+    // Tracking pattern matches for a given Rule.
+    class RulePatternMatch {
+    public:
+        RulePatternMatch(Rule* rule, const u_char* data, int data_len, MatchPos end_of_match)
+            : rule(rule), text(new String(data, data_len, false)), end_of_match(end_of_match) {}
```

Comment:
Suggest to use `unique_ptr` for `text` to avoid manual memory mgmt.

Reply:
Or, actually, do we need this on the heap at all, vs. just storing a `String` directly inside the `class`?